hydra/_internal/core_plugins/fish_completion.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+import os
+import sys
+from typing import List, Optional, Tuple
+
+from hydra.plugins.completion_plugin import CompletionPlugin
+
+log = logging.getLogger(__name__)
+
+
+class FishCompletion(CompletionPlugin):
+    """Issue 1: python script.py style rule cannot be uninstalled
+    Issue 2: fish adds a space to "hydra. " automatically"""
+
+    def install(self) -> None:
+        output = self._get_exec()
+        script = []
+        for name, cond, exe in output:
+            args = f"'({exe} -sc query=fish --_completion (commandline -cp))'"
+            script.append(f"""complete -c {name} {cond}-x -a {args}\n""")
+        print("".join(script))
```

Comment:
What does the install command looks like (actual output) ?

Reply:
Somehow I can't find the reply box for your previous comment about registering completion function. I think it's technically doable to mimic what has been done for bash. I actually thought of reusing that solution, but fish is not POSIX compliant or bash compatible. It has a different scripting syntax and I think it makes the solution more complex and less maintainable.

Sorry I'm caught up in other things today and tmr. Will come back to address your other comments after.

Reply:
we should definitely not try to reuse the same function from bash, it's super specific. but the concept of having a fish specific completion function installed seems good to me.
that function would be able to bridge the gap between what Hydra is expecting and what fish is asking for.

complete --help has this, it might be what I am after.
```
       complete -c rpm -n '__fish_contains_opt -s e erase' -d nodeps 'Don't check dependencies'

       where __fish_contains_opt is a function that checks the command line
       buffer for the presence of a specified set of options.
```

Reply:
I'm not sure I understand what you mean by the help info.